### PR TITLE
Extras

### PR DESCRIPTION
**Pull request by benlaurie**
Add a variant on STL generation.

### CODE REVIEW

File: include/algo_helpers.h [hunk 1]
```diff
@@ -126,6 +127,8 @@ template<class Vector>
 float min_seg_dist(const Vector &B0, const Vector &B0p, const Vector &B1,
                    const Vector &B1p,float &s,float &t);
 
+void write_STL(ostream &file, const Tube<Vector3> &tube, const int S);
```
Comment (mcarlen) [inline include/algo_helpers.h]: My very vague recollection of this file that it contained helper things to run simulations, not really I/O aspects of curves/tubes.

Apparently most of the read*, write* methods are in Tube/Curve directly. Just move writeSTL into Tube?

File: tools/xyz2pkf.cpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 
-#include "../include/Curve.h"
+#include <Curve.h>
```
Comment (mcarlen) [inline tools/xyz2pkf.cpp:10]: Just curious, why the change away from relative here?

File: tools/pkf2stl2.cpp [hunk 3]
```diff
@@ -27,6 +27,24 @@ static void write_facet(ostream &file,
        << el << ef;
 }
 
+static void write_STL(ostream &file, const Tube<Vector3> &tube, const int S) {
```
Comment (mcarlen) [inline tools/pkf2stl2.cpp]: I have seen this one before. Move it into Tube and call it here instead?

File: tools/pkf2stl2.cpp [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  \brief Read data from a PKF file, produce a tubular mesh for each
+  	curve in the file and write that to an STL file.
+*/
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+
+#include <TubeBundle.h>
+#include <algo_helpers.h>
+
+int main(int argc, char **argv) {
```
Comment (mcarlen) [inline tools/pkf2stl2.cpp]: This could probably have been absorbed as an additional arg to pkf2stl, but this works too.

Mind adding a comment here as to why there is a pkf2stl and a pkf2stl2?

File: tools/xyz2pkf.cpp (hunk 2 above)
Comment (benlaurie) [reply tools/xyz2pkf.cpp:10]: No particularly great reason, but it seems to me you should either use `-I../include` or `"../include/..."`, but not both...

File: tools/pkf2stl2.cpp (hunk 3 above)
Comment (benlaurie) [reply tools/pkf2stl2.cpp]: Yes, this was removed in a later commit so will be fixed at the same time as the other one.

File: tools/pkf2stl2.cpp (hunk 4 above)
Comment (benlaurie) [reply tools/pkf2stl2.cpp]: Perhaps I will merge them now its clearer how they differ...

File: include/algo_helpers.h (hunk 1 above)
Comment (benlaurie) [reply include/algo_helpers.h]: Done.

File: tools/pkf2stl2.cpp (hunk 4 above)
Comment (mcarlen) [reply tools/pkf2stl2.cpp]: In this patch or a follow up?

Comment (benlaurie) [reply tools/pkf2stl2.cpp]: Done
